sound/soc/sof/ipc3-topology.c

```diff
@@ -2241,7 +2241,7 @@ static int sof_tear_down_left_over_pipelines(struct snd_sof_dev *sdev)
 		for_each_pcm_streams(dir) {
 			struct snd_pcm_substream *substream = spcm->stream[dir].substream;
 
-			if (!substream || !substream->runtime)
+			if (!substream || !substream->runtime || spcm->stream[dir].suspend_ignored)
```

Comment:
I still find the commit message very convoluted with 4 to 5 line sentences combining multiple propositions. I would recommend starting with IPC4 requirements and explaining what we do. I also didn't see why we special case the suspend when it's really an across the board problem. It feels like the suspend case is how we found the problem, but it's only a symptom of a more general issue?

Reply:
@plbossart we didnt identify the problem because of the suspend case. For IPC4, we are not triggering the pipelines in the right order as needed by the firmware today and we need to switch them around. But switching the order is not so easy because of the special case handling for the suspend trigger. So, I eliminate the special handling for suspend in the first patch and then switch the order.